File: app/src/main/kotlin/jb/openware/app/components/FolderManagement.java
package in.afi.codekosh.components;

import android.os.Environment;

public class FolderManagement {

    private static final String[] FOLDER_NAMES = {
            "Download/CodeKosh",
            "Documents/CodeKosh",
            "Download/CodeKosh/Sketchware",
            "Download/CodeKosh/Sketchware/temp",
            "Download/CodeKosh/Sketchware/.temp",
            "Download/CodeKosh/Sketchware Pro",
            "Download/CodeKosh/Android Studio",
            "Download/CodeKosh/HTML",
            "Documents/CodeKosh/Sketchware Projects",
            "Documents/CodeKosh/Sketchware Projects/Export"
    };

    public FolderManagement() {
    }

    public void makeFolders() {
        for (String folderName : FOLDER_NAMES) {
            String folderPath = Environment.getExternalStorageDirectory() + "/" + folderName;
            if (!FileUtil.isExistFile(folderPath)) {
                FileUtil.makeDir(folderPath);
            }
        }
    }
}

File: app/src/main/kotlin/jb/openware/app/ui/components/FolderManagement.kt
package jb.openware.app.ui.components

import android.os.Environment
import jb.openware.app.util.FileUtil

class FolderManagement {

    companion object {
        private val FOLDER_NAMES = listOf(
            "Download/OpenWare",
            "Download/OpenWare/Apk",
        )
    }

    fun makeFolders() {
        val basePath = Environment.getExternalStorageDirectory().absolutePath

        FOLDER_NAMES.forEach { name ->
            val path = "$basePath/$name"
            if (!FileUtil.isExistFile(path)) {
                FileUtil.makeDir(path)
            }
        }
    }
}
